cmd/login/login.go

```diff
@@ -0,0 +1,97 @@
+package login
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/davidebianchi/go-jsonclient"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+const miactlAppID = "miactl"
+
+type tokenRequest struct {
+	GrantType  string `json:"grant_type"`
+	Username   string `json:"username"`
+	Password   string `json:"password"`
+	AppID      string `json:"appId"`
+	ProviderID string `json:"providerId"`
+}
+
+type tokenResponse struct {
+	AccessToken  string `json:"accessToken"`
+	RefreshToken string `json:"refreshToken"`
+	ExpireAt     int64  `json:"expiresAt"`
+}
+
+// NewLoginCmd create a new Login command
+func NewLoginCmd() *cobra.Command {
+	var (
+		username   string
+		password   string
+		providerID string
+	)
+
+	cmd := &cobra.Command{
+		Use:   "login",
+		Short: "authenticate with console",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			baseURL := viper.GetString("apibaseurl")
+			if baseURL == "" {
+				return errors.New("API base URL not specified nor configured")
+			}
+
+			accessToken, err := login(baseURL, username, password, providerID)
+			if err != nil {
+				return err
+			}
+
+			// save current token for later commands
+			viper.Set("apitoken", accessToken)
+			fmt.Println("OK")
+			return viper.WriteConfig()
```

Comment:
If write config fails, user read OK and after the error stack trace.

Should we invert the command: before write config and later the ok message?

Reply:
Yes, that would be definitely better.